packages/framer-motion/src/render/dom/DOMKeyframesResolver.ts

```diff
@@ -0,0 +1,183 @@
+         * temporarily set styles to the destination keyframes.
+         * Skip if we have more than two keyframes or this isn't a positional value.
+         * TODO: We can throw if there are multiple keyframes and the value type changes.
+         */
+        if (!positionalKeys.has(name) || unresolvedKeyframes.length !== 2) {
+            return
+        }
+
+        const [origin, target] = unresolvedKeyframes
+        const originType = findDimensionValueType(origin)
+        const targetType = findDimensionValueType(target)
+
+        /**
+         * Either we don't recognise these value types or we can animate between them.
+         */
+        if (!originType || !targetType || originType === targetType) return
+
+        /**
+         * If both values are numbers or pixels, we can animate between them by
+         * converting them to numbers.
+         */
+        if (isNumOrPxType(originType) && isNumOrPxType(targetType)) {
+            for (let i = 0; i < unresolvedKeyframes.length; i++) {
+                const value = unresolvedKeyframes[i]
+                if (typeof value === "string") {
+                    unresolvedKeyframes[i] = parseFloat(value as string)
+                }
+            }
+        } else {
+            /**
+             * Else, the only way to resolve this is by measuring the element.
+             */
+            this.needsMeasurement = true
+        }
+    }
+
+    unsetTransforms() {
+        const { element, name, unresolvedKeyframes } = this
+
+        if (!element.current) return
+
+        this.removedTransforms = removeNonTranslationalTransform(element)
+
+        const finalKeyframe =
+            unresolvedKeyframes[unresolvedKeyframes.length - 1]
+
+        element.getValue(name, finalKeyframe).jump(finalKeyframe, false)
+    }
+
+    measureInitialState() {
+        const { element, unresolvedKeyframes, name } = this
+
+        if (!element.current) return
+
+        if (name === "height") {
+            this.suspendedScrollY = window.pageYOffset
+        }
+
+        this.measuredOrigin = positionalValues[name](
+            element.measureViewportBox(),
+            window.getComputedStyle(element.current)
+        )
+
+        unresolvedKeyframes[0] = this.measuredOrigin
+    }
+
+    renderEndStyles() {
+        this.element.render()
+    }
+
+    measureEndState() {
+        const { element, name, unresolvedKeyframes } = this
+
+        if (!element.current) return
+
+        const value = element.getValue(name)
+        value && value.jump(this.measuredOrigin, false)
+
+        unresolvedKeyframes[unresolvedKeyframes.length - 1] = positionalValues[
+            name
+        ](
+            element.measureViewportBox(),
+            window.getComputedStyle(element.current)
+        ) as any
+
+        if (name === "height" && this.suspendedScrollY !== undefined) {
+            window.scrollTo(0, this.suspendedScrollY)
```

Comment:
Why do we need to scroll the window to 0?

Reply:
We're scrolling it to the value it was before measurement - the measurement can fuck with page scroll